MediaBrowser.Model/Dlna/StreamBuilder.cs

```diff
@@ -803,23 +807,28 @@ private StreamInfo BuildVideoItem(MediaSourceInfo item, MediaOptions options)
             string? audioCodec)
         {
             // Prefer matching video codecs
-            var videoCodecs = ContainerProfile.SplitValue(videoCodec);
+            var videoCodecs = new List<string>();
+            if (videoStream is not null && videoCodec.ContainsContainer(videoStream.Codec))
+            {
+                videoCodecs.Add(videoStream.Codec!);
+            }
 
-            // Enforce HLS video codec restrictions
-            if (string.Equals(playlistItem.SubProtocol, "hls", StringComparison.OrdinalIgnoreCase))
+            if (videoCodec is not null)
             {
-                videoCodecs = videoCodecs.Where(codec => _supportedHlsVideoCodecs.Contains(codec)).ToArray();
+                var splitString = videoCodec.SpanSplit(',');
+                while (splitString.MoveNext())
+                {
+                    videoCodecs.Add(splitString.Current.ToString());
```

Comment:
Just use .Split. You would have to benchmark whether enumerating is actually more time and memory saving than simply allocating an array and adding it to the list. I am guessing that it isn't worth the added complexity.